runtime/src/accounts_db.rs

```diff
@@ -298,12 +298,18 @@ impl AccountStorageEntry {
             status = AccountStorageStatus::Available;
         }
 
-        if count > 0 {
-            *count_and_status = (count - 1, status);
-        } else {
-            warn!("count value 0 for slot {}", self.slot_id);
-        }
-        count_and_status.0
+        // Some code path is removing accounts too many; this may result in an
+        // unintended reveal of old state for unrelated accounts.
```

Comment:
This actually happened locally in the way-before past locally.

It's very likely if this ever happens for this to cause the disastrous event of ledger inconsistency.